tests/test_reco_diversity.py

```diff
@@ -0,0 +1,56 @@
+# -*- coding: utf-8 -*-
+# Copyright FMR LLC <opensource@fidelity.com>
+# SPDX-License-Identifier: Apache-2.0
+
+import unittest
+
+import pandas as pd
+
+from jurity.recommenders import DiversityRecoMetrics
+from jurity.utils import Constants
+
+
+class TestDiversityRecommenders(unittest.TestCase):
+
+    def test_inter_list_diversity(self):
+        actual = pd.DataFrame({Constants.user_id: [0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2],
+                               Constants.item_id: [0, 1, 2, 3,
+                                                   0, 1, 2, 4,
+                                                   0, 1, 2, 5],
+                               'score': [1, 1, 1, 1,
+                                         0, 0, 0, 0,
+                                         0, 0, 1, 1]})
+        predicted = pd.DataFrame({Constants.user_id: [0, 0, 0, 0, 1, 1, 1, 1, 2, 2, 2, 2],
+                                  Constants.item_id: [0, 1, 2, 3,
+                                                      0, 1, 2, 4,
+                                                      0, 1, 2, 5
+                                                      ],
+                                  'score': [0.9, 0.7, 0.6, 0.3,
+                                            0.9, 0.7, 0.4, 0.1,
+                                            0.9, 0.8, 0.6, 0.6
+                                            ]
+                                  })
+
+        metric = DiversityRecoMetrics.InterListDiversity(click_column='score', k=4)
+        results = metric.get_score(actual, predicted, batch_accumulate=False, return_extended_results=True)
+
+        self.assertEqual(results['inter-list diversity'], 0.25)
```

Comment:
this is a nitpicking comment (OK if you discard) but I prefer "unit tests" as in each test tests a specific behavior. 

This one tests for a few different cases all at once.. when any of it fails, the whole fails which is a bit misleading

That means duplicating the input data etc. but at the end we have individual tests that can run standalone..

Plus, with some meaningful names for each unit test, it is easy to see what we are checking. e.g., test_sample_size, test_chunk_size etc.. Here I see one giant test but I need read code to understand what it is really checking ;)

Reply:
Agree. It makes sense to split these tests with each one serving a different purpose. And it will be easier to capture what each test is doing.